### PR TITLE
Showing proper error while creating a csr ndarray which is not 2D in shape.

### PR DESCRIPTION
**Pull request by piyushghai**
## Description ##
Fixes (partially) https://github.com/apache/incubator-mxnet/issues/13193. 
We were allowing non 2D NDArrays to be created as a CSR that caused inconsistencies and also caused a crash when one tried to access ```.data``` on a CSR Matrix. The crash was coming from here : https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/cast_storage-inl.h#L238. 
The other places constructing CSRNDArray are already guarded by the shape checks. 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: tests/python/unittest/test_sparse_ndarray.py [hunk 1]
```diff
@@ -962,6 +962,11 @@ def test_sparse_nd_check_format():
     indptr_list = [0, -2, 2, 3]
     a = mx.nd.sparse.csr_matrix((data_list, indices_list, indptr_list), shape=shape)
     assertRaises(mx.base.MXNetError, a.check_format)
+    # CSR format should be 2 Dimensional.
+    a = mx.nd.array([1, 2, 3]).tostype('csr')
+    assertRaises(mx.base.MXNetError, a.check_format)
```
Comment (eric-haibin-lin) [inline tests/python/unittest/test_sparse_ndarray.py]: I think you want to do `a = mx.nd.array([1, 2, 3]); assertRaises(ValueError, a.tostype, 'csr') `instead?

**Comment by eric-haibin-lin** [comment]
Does this fix the elemwise_mul issue?

File: tests/python/unittest/test_sparse_ndarray.py (hunk 1 above)
Comment (piyushghai) [reply tests/python/unittest/test_sparse_ndarray.py]: Will make the change.

**Comment by piyushghai** [comment]
> Does this fix the elemwise_mul issue?

In that single issue there are two issues essentially, and both are independent of each other. 
I'll open a separate issue to track the ```elemwise_mul operator``` issue. Will post the issue link here. 
Currently swamped with other tasks, so will get back on this early next week :)

**Comment by roywei** [comment]
@piyushghai could you take a look at the CI failure? thanks!

**Comment by abhinavs95** [comment]
@piyushghai Gentle ping..

**Comment by piyushghai** [comment]
@mxnet-label-bot Update [pr-work-in-progress]

**Comment by piyushghai** [comment]
Closing this. 
Will reopen a new one after raising a github issue for the error message.

**Pull request closed without merging**